### PR TITLE
Set User-Agent header on both REST and graphql requests

### PR DESCRIPTION
**Pull request by yob**
In #256 we starting setting a custom User-Agent header on REST API calls, to help us understand the ways our API is used. It looks like this:

    Terraform/ (+https://www.terraform.io) Terraform-Plugin-SDK/2.0.3 buildkite/0.15.0

This builds on that work and sets the same User-Agent on graphql API requests as well.

It drops the unnecessary use of golang.org/x/oauth2 to set a static bearer token header, and uses the [tripperware pattern](https://dev.to/stevenacoffman/tripperwares-http-client-middleware-chaining-roundtrippers-3o00) to set both headers we care about:

1. the static bearer token for auth
2. the user agent

The same http.Client is used for both REST and graphql, so we'll get identical behaviour on both APIs.

### CODE REVIEW

**Comment by yob** [comment]
@jradtilbrook @mcncl absolutely no hurry on this one, I just thought I'd poke at it while the changes in #256 were fresh in our brains. If we merge it, it could wait for the next release you do for other reasons.

File: buildkite/client.go
Comment (danstn) [inline buildkite/client.go:72]: Named returns are often considered harmful, also not used here.

```suggestion
func (rt *headerRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
```

File: buildkite/client.go
Comment (danstn) [inline buildkite/client.go:70]: Linter yelling continues :) Local bindings [should be](https://go-critic.com/overview.html#captlocal) lower-case.

```suggestion
func newHeaderRoundTripper(next http.RoundTripper, header http.Header) *headerRoundTripper {
	if next == nil {
		next = http.DefaultTransport
	}
	return &headerRoundTripper{
		next:   next,
		Header: header,
	}
}
```

File: buildkite/client.go
Comment (danstn) [inline buildkite/client.go:78]: `RoundTrip` returns an error, so normally we'd want to wrap it to provide additional context.

**Review by danstn — DISMISSED** [review]
Great idea @yob! 

There are a few minor comments, but I think it's safe to skip these and add https://github.com/golangci/golangci-lint in a separate PR instead.

**Comment by mcncl** [comment]
Happy to merge once @danstn suggestions are committed in. Also happy to just commit them from comment too.

**Comment by danstn** [comment]
> Happy to merge once @danstn suggestions are committed in. Also happy to just commit them from comment too.

Oh these are not blocking, happy to land it as is. @yob up to you

**Comment by yob** [comment]
They're good suggestions, I'll make them in a bit :heart:

**Review by mcncl — APPROVED** [review]
LGTM

File: buildkite/client.go
Comment (yob) [reply buildkite/client.go:77]: I've address the other feedback, but left this one. I could wrap the error, but... I really wasn't sure what to add that was meaningful.

**Comment by yob** [comment]
I've confirmed that we see the Terraform useragent in our API access logs for both the REST **and** graphql calls

**Pull request merged**
